### ISSUE TITLE
Add error for passing stdin to builtins which do not expect it

### ISSUE THREAD

**Issue by alexjurkiewicz** [koalaman/shellcheck#921]
I just spent an embarrassing amount of time troubleshooting the following snippet:

```sh

#!/bin/bash

echo >foo <<EOF
hello
EOF

```

Of course, I really meant to use `cat` rather than `echo`. shellcheck doesn't give any error for this file, but I think it could detect if you are passing something on stdin to a builtin that doesn't expect it. Example builtins would be `echo`, `source`, `alias`, etc etc.

#### Here's what shellcheck currently says:

No error.

#### Here's what I wanted or expected to see:

SCxxxx: 'echo' builtin does not expect stdin - is this a mistake?

**Comment by koalaman** [comment]
This functionality has now been added for both pipes and redirections to simple commands (compound commands pending). Thanks!

**Comment by alexjurkiewicz** [comment]
Wow, nice! 🙇
